### PR TITLE
Consolidate robustness github actions templates

### PR DESCRIPTION
**Pull request by jmhbnz**
This pull request consolidates our existing `robustness-template.yaml` and `robustness-template-arm64.yaml` actions re-usable workflow templates into a single file.

The reason these were originally separated is `arm64` robustness jobs were configured to run within containers. As this was a new approach we did this separately initially only for `arm64` workflows.

Now these have been running successfully for a while we have confidence that running the jobs in containers is working as expected so we should consolidate the workflow templates once more to run all robustness jobs within containers.

This is progress towards https://github.com/etcd-io/etcd/issues/15956.

### CODE REVIEW

File: .github/workflows/robustness-nightly.yaml [hunk 1]
```diff
@@ -31,12 +31,13 @@ jobs:
...
       artifactName: release-35
   release-35-arm64:
-    uses: ./.github/workflows/robustness-template-arm64.yaml
+    uses: ./.github/workflows/robustness-template.yaml
     with:
       etcdBranch: release-3.5
       count: 100
       testTimeout: 200m
       artifactName: release-35-arm64
+      runs-on: "['self-hosted', 'Linux', 'ARM64']"
```
Comment (jmhbnz) [inline .github/workflows/robustness-nightly.yaml:40]: This is a major omission in one of my previous pull requests! 🤦🏻‍♂️

File: .github/workflows/robustness-template.yaml [hunk 2]
```diff
@@ -21,12 +21,22 @@ on:
         type: string
         default: "['ubuntu-latest']"
 permissions: read-all
+
 jobs:
   test:
     timeout-minutes: 210
     runs-on: ${{ fromJson(inputs.runs-on) }}
+    container:
```
Comment (jmhbnz) [inline .github/workflows/robustness-template.yaml:29]: Running the `amd64` robustness workflows in containers does not appear to have negatively impacted runtime.

I ran workfows `on: pull_request` initially to validate. Workflows passed and runtimes were about the same:

- Original: https://github.com/etcd-io/etcd/actions/runs/5840581040
- This pr: https://github.com/etcd-io/etcd/actions/runs/5845356933

**Review by ahrtr — APPROVED** [review]
LGTM

thx @jmhbnz

**Comment by ahrtr** [comment]
@jmhbnz please rebase this PR

**Pull request merged**